File: backend/hospital-service/src/main/java/com/bharatteeka/hospital/service/SlotService.java
package com.bharatteeka.hospital.service;

import com.bharatteeka.hospital.dto.SlotRequest;
import com.bharatteeka.hospital.entity.Slot;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.List;

public interface SlotService {

    List<Slot> getSlotsByDateAndHospital(LocalDate date, Integer hospitalId);

    List<Slot> getSlotsByDateAndTime(Integer hospitalId, LocalDate date, LocalTime time);

    List<Slot> getAvailableSlots(Integer hospitalId, LocalDate date);

    List<Slot> getSlotsByVaccine(Integer hospitalId, Integer vaccineId, LocalDate date);
    
    Slot createSlot(SlotRequest request);

    Slot updateSlot(Integer slotId, SlotRequest request);

    void deleteSlot(Integer slotId);

}

File: frontend/src/services/slotService.js
import { hospitalApi } from "./apiClients";

const slotService = {
  createSlot: (slotData) => hospitalApi.post("/hospital/slots", slotData),

  updateSlot: (slotId, slotData) =>
    hospitalApi.put(`/hospital/slots/${slotId}`, slotData),

  deleteSlot: (slotId) => hospitalApi.delete(`/hospital/slots/${slotId}`),

  getAllSlots: () => hospitalApi.get("/hospital/slots"),

  getSlotsByHospital: (hospitalId) =>
    hospitalApi.get(`/hospital/slots/hospital/${hospitalId}`),

  getSlotsByHospitalAndDate: (hospitalId, date) =>
    hospitalApi.get("/hospital/slots", { params: { hospitalId, date } }),

  getAvailableSlots: (hospitalId, date) =>
    hospitalApi.get(`/hospital/slots/available/${hospitalId}`, {
      params: { date },
    }),
};

export default slotService;
